internal/util/search/search_test.go

```diff
@@ -0,0 +1,76 @@
+package search
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type patternTest struct {
+	name         string
+	fieldValue   string
+	valueRegex   string
+	patternRegex string
+	expected     string
+	errMsg       string
+}
+
+var resolvePatternCases = []patternTest{
+	{
+		name:         "resolve pattern 1",
+		fieldValue:   "foo1-dev-bar1-us-east-1-baz1",
+		valueRegex:   "([a-z]+[0-9]+)-dev-([a-z]+[0-9]+)-us-east-1-([a-z]+[0-9]+)",
```

Comment:
nit: you can use metacharacter `\w` for brevity

Reply:
Done.